plugin/cache/handler.go

```diff
@@ -172,4 +171,4 @@ func setDo(m *dns.Msg) {
 
 // defaultUDPBufsize is the bufsize the cache plugin uses on outgoing requests that don't
 // have an OPT RR.
-const defaultUDPBufSize = 2048
+const defaultUDPBufSize = 512
```

Comment:
Do we need both changes?  I'd be inclined to change just `defaultUDPBufSize` and leave `setDo` unchanged.

Reply:
We (specifically we as in openshift) do not need both changes, we could just change this `defaultUDPBufSize`. The pro of removing `o.SetUDPSize(defaultUDPBufSize)` is that if `bufsize` ever changes in the Corefile, upstream requests will use `bufsize`.

Openshift always use 512, and you can't change the corefile, so you could argue we could just change this default. _And the other argument is:_ users can't configure upstream bufsize with CoreDNS at all today...so we aren't really regressing anything either way we do it.

I think it would be slightly safer to just change `defaultUDPBufSize` to 512 to limit the impact of the changes.

Reply:
Fair enough.  We don't support changing `bufsize` in Corefile, but there's no good reason to override `bufsize` gratuitously for people who are modifying the Corefile for diagnostic purposes or unsupported use-cases.

Reply:
To be clear, I agree with including the change to `setDo` in addition to changing `defaultUDPBufSize`.

Reply:
Sure, good discussion. We are adding a tiny bit of flexibility with the change in `setDo`, which I think we might-as-well give users for testing/unsupported use-cases, but does increase the blast-radius a tiny bit.